src/utils/polling.js

```diff
@@ -114,16 +153,32 @@ async function pollSubscriptionTopics() {
         };
         db.upsert('polling', poll, (p) => p.resource === resourceToPoll);
 
+        const newResources = [];
+        const modifiedResources = [];
+
         // Store fetched resources in local database
         if (data.total > 0) {
           logger.info(`Storing ${data.total} fetched resource(s) for ${resourceToPoll} into database.`);
           const resources = data.entry.map(entry => entry.resource);
 
           resources.forEach((resource) => {
             const collection = `${resource.resourceType.toLowerCase()}s`;
-            db.insert(collection, resource);
+
+            // Determine if resource is new before inserting
+            const storedResource = db.select(collection, r => r.id === resource.id);
+            if (storedResource.length === 0) newResources.push(resource);
+            else modifiedResources.push(resource);
+
+            db.upsert(collection, { id: resource.id }, r => r.id === resource.id);
           });
         }
+
+        // Don't send notifications on first poll
```

Comment:
This works but I'm wondering if this will be an issue if the polling time is really long (i.e. not 1 minute). I feel like the there should be a "poll" as soon as we decide to start polling for a resource. This would fetch the base resources but will not send a notification. Whenever the next poll is scheduled it will then always send notifications as appropriate

Reply:
When do we decide to start polling for a resource? Should we immediately poll when a subscription is added that has not already been polled for?

Reply:
So I was thinking there was some set up when a subscription is added but there is not. Let's see what @dehall thinks

Reply:
Yeah that's s good point. I think it's cleaner to do an initial "pull" when the subscription is first setup and then the logic here is the same whether it's the first or 10000th run. I think that would just involve copying a few lines from here to the initial setup logic, so any objections to making that in this pr?

Reply:
Should be done!